src/preact/hooks/useLocalStorage.js

```diff
@@ -0,0 +1,36 @@
+import {useCallback, useState} from '#preact';
+
+/**
+ * @param {string} key
+ * @param {any} [defaultValue]
+ * @return {{ "0": any, "1": function(any): void }}
+ */
+export function useLocalStorage(key, defaultValue = null) {
+  // Keep track of the state locally:
+  const [value, setValue] = useState(() => {
+    try {
+      // eslint-disable-next-line local/no-forbidden-terms
+      const json = self.localStorage?.getItem(key);
+      return json ? JSON.parse(json) : defaultValue;
```

Comment:
should this be `??`? in the case that consumers purposely store `''` and expect to get that back instead of the defaultValue?

Reply:
Everything stored is json-encoded, so storing an empty string will store `'""'`.  So I think this should work as expected.